cumulusci/tasks/bulkdata/step.py

```diff
@@ -220,6 +232,32 @@ def get_results(self):
                 yield from reader
 
 
+class RestApiQueryOperation(BaseQueryOperation):
+    """Operation class for REST API query jobs."""
+
+    def __init__(self, *, sobject, fields, api_options, context, query):
+        super().__init__(
+            sobject=sobject, api_options=api_options, context=context, query=query
+        )
+        self.fields = fields
+
+    def query(self):
+        # This is not as memory-efficient as we might like.
+        # To preserve the invariant from the Bulk API that
+        # we know the query count before consuming the results,
+        # we use `query_all` instead of `query_all_iter`.
```

Comment:
Couldn't you get the first batch with `query` and grab the `totalSize`, and then make get_results call `query_more` repeatedly the same way that `query_all_iter` does?

Reply:
I was shying away from adding more complexity, but turns out to be quite straightforward - thanks for the hint.